### PR TITLE
script: Use FLoat32Array in XRRay

### LINKED ISSUE #31048: WebIDL: use FLoat32Array in XRRay
Depends on https://github.com/servo/servo/pull/30990

Essentially the same as https://github.com/servo/servo/issues/31047

### PR DESCRIPTION
**Pull request by Taym95**
---

- [X] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [X] These changes fix #31048

- [x] These changes do not require tests because not changing behavior

### CODE REVIEW

File: components/script/dom/xrray.rs [hunk 1]
```diff
@@ -153,16 +149,21 @@ impl XRRayMethods for XRRay {
...
-            create_typed_array(cx, &arr, &self.matrix);
+            self.matrix
+                .set_data(_cx, &arr)
+                .expect("failed to set projection matrix")
         }
-        NonNull::new(self.matrix.get()).unwrap()
+
+        self.matrix
+            .get_internal()
+            .expect("failed to get projection matrix")
```
Comment (Taym95) [inline components/script/dom/xrray.rs:167]:
```suggestion
            .expect("Failed to get transform's internal matrix.")
```

**Review by gterzian — APPROVED** [review]
LGTM with a few nits, and thanks for the step renumbering!

**Comment by gterzian** [comment]
This one can merge once the conflict is resolved.

File: components/script/dom/xrray.rs [hunk 2]
```diff
@@ -153,16 +149,21 @@ impl XRRayMethods for XRRay {
...
+            // Step 6
             // According to the spec all matrices are column-major,
             // however euclid uses row vectors so we use .to_array()
             let arr = RigidTransform3D::new(rotation, translation)
                 .to_transform()
                 .to_array();
-            create_typed_array(cx, &arr, &self.matrix);
+            self.matrix
+                .set_data(_cx, &arr)
+                .expect("Failed to set on data on the xrray's matrix.")
```
Comment (mrobinson) [inline components/script/dom/xrray.rs:162]:
```suggestion
                .expect("Failed to set matrix data on XRRAy.")
```

File: components/script/dom/xrray.rs [hunk 3]
```diff
@@ -153,16 +149,21 @@ impl XRRayMethods for XRRay {
...
         }
-        NonNull::new(self.matrix.get()).unwrap()
+
+        self.matrix
+            .get_internal()
+            .expect("Failed to get the xrray's matrix.")
```
Comment (mrobinson) [inline components/script/dom/xrray.rs:167]:
```suggestion
            .expect("Failed to get matrix from XRRay.")
```

**Review by mrobinson — APPROVED** [review]
Just a couple tiny nits. I'll apply these and land this!

**Pull request merged**